### ISSUE TITLE
Map get sometimes segfaults

### ISSUE THREAD

**Issue by cpcloud** [duckdb/duckdb#6088]
### What happens?

Accessing keys of a map column segfaults

```
D select map(list_value(string_col), list_value(int_col))[string_col] from read_csv_auto('/home/cloud/src/ibis/ci/ibis-testing-data/functional_alltypes.csv') where string_col <> '0' limit 10;
zsh: segmentation fault (core dumped)  build/release/duckdb
```

### To Reproduce

The follow query non-deterministically, but consistently after around 5-15 tries reproduces the error:

```sql
SELECT
  MAP(LIST_VALUE(string_col), LIST_VALUE(int_col))[string_col]
FROM READ_CSV_AUTO('/home/cloud/src/ibis/ci/ibis-testing-data/functional_alltypes.csv')
WHERE
  string_col <> '0'
LIMIT 10
```

### OS:

NixOS

### DuckDB Version:

v0.6.2-dev2059 67dab2a30b

### DuckDB Client:

CLI

### Full Name:

Phillip Cloud

### Affiliation:

Voltron Data

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by cpcloud** [comment]
The first 10 rows + header of the referenced CSV file are:

```
index,"Unnamed: 0",id,bool_col,tinyint_col,smallint_col,int_col,bigint_col,float_col,double_col,date_string_col,string_col,timestamp_col,year,month
0,0,6690,1,0,0,0,0,0.0,0.0,11/01/10,0,"2010-11-01 00:00:00.000",2010,11
1,1,6691,0,1,1,1,10,1.10000002384,10.1,11/01/10,1,"2010-11-01 00:01:00.000",2010,11
2,2,6692,1,2,2,2,20,2.20000004768,20.2,11/01/10,2,"2010-11-01 00:02:00.100",2010,11
3,3,6693,0,3,3,3,30,3.29999995232,30.3,11/01/10,3,"2010-11-01 00:03:00.300",2010,11
4,4,6694,1,4,4,4,40,4.40000009537,40.4,11/01/10,4,"2010-11-01 00:04:00.600",2010,11
5,5,6695,0,5,5,5,50,5.5,50.5,11/01/10,5,"2010-11-01 00:05:00.100",2010,11
6,6,6696,1,6,6,6,60,6.59999990463,60.6,11/01/10,6,"2010-11-01 00:06:00.150",2010,11
7,7,6697,0,7,7,7,70,7.69999980927,70.7,11/01/10,7,"2010-11-01 00:07:00.210",2010,11
8,8,6698,1,8,8,8,80,8.80000019073,80.8,11/01/10,8,"2010-11-01 00:08:00.280",2010,11
9,9,6699,0,9,9,9,90,9.89999961853,90.9,11/01/10,9,"2010-11-01 00:09:00.360",2010,11
```

**Comment by LindsayWray** [comment]
This issue should be fixed now (see #6109), thanks for reporting!
